js/actions/tests/loadSourceText.js

```diff
@@ -23,20 +29,111 @@ const mockThreadClient = {
   }
 };
 
+/**
+ * Useful for testing async actions where
+ * start and error states matter.
+ */
+const deferredMockThreadClient = {
+  request: undefined,
+  source: function(form) {
+    return {
+      source: () => {
+        let deferred = promise.defer();
+        this.request = deferred;
+        return deferred.promise;
+      }
+    };
+  },
+  getRequest: function() {
+    return this.request;
+  }
+};
+
 describe("loadSourceText", () => {
-  beforeEach(function() {
-    this.store = createStore(mockThreadClient);
+  describe("loading one source text", function() {
+    beforeEach(function(done) {
+      this.store = createStore(simpleMockThreadClient);
+      this.store.dispatch(actions.loadSourceText({ actor: "foo1" }))
+        .then(() => done());
+    });
+
+    it("Store has the source text", function() {
+      const fooSourceText = queries.getSourceText(
+                              this.store.getState(), "foo1");
+      expect(fooSourceText.get("text")).to.equal(sourceText.foo1.source);
+    });
   });
 
-  describe("loading one source text", function() {
+  describe("loading two different sources", function() {
     beforeEach(function(done) {
+      this.store = createStore(simpleMockThreadClient);
       this.store.dispatch(actions.loadSourceText({ actor: "foo1" }))
+        .then(() => {
+          return this.store.dispatch(actions.loadSourceText({ actor: "foo2" }));
+        })
         .then(() => done());
     });
 
+    it("Store has the first source text", function() {
+      const fooSourceText = queries.getSourceText(
```

Comment:
fwiw, if you use a query a lot in a test, you can still destructure it at the top (`const { getSourceText } = queries`).